unity-client/Assets/Scripts/MainScripts/DCL/Controllers/MapRenderer/MapRenderer.cs

```diff
@@ -2,33 +2,118 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.UI;
+using TMPro;
 
 namespace DCL
 {
     public class MapRenderer : MonoBehaviour
     {
+        const int LEFT_BORDER_PARCELS = 25;
+        const int RIGHT_BORDER_PARCELS = 31;
+        const int TOP_BORDER_PARCELS = 31;
+        const int BOTTOM_BORDER_PARCELS = 25;
+        const int WORLDMAP_WIDTH_IN_PARCELS = 300;
+
         public static MapRenderer i { get; private set; }
 
+        [SerializeField] private float parcelHightlightScale = 1.25f;
+        private float parcelSizeInMap;
         private Vector3Variable playerWorldPosition => CommonScriptableObjects.playerWorldPosition;
         private Vector3Variable playerRotation => CommonScriptableObjects.cameraForward;
+        private Vector3[] mapWorldspaceCorners = new Vector3[4];
+        private Vector3 worldCoordsOriginInMap;
 
+        [HideInInspector] public Vector3 mouseMapCoords;
         public Vector3 playerGridPosition => Utils.WorldToGridPositionUnclamped(playerWorldPosition.Get());
-
         public MapAtlas atlas;
+        public RawImage parcelHighlightImage;
+        public TextMeshProUGUI highlightedParcelText;
         public Transform overlayContainer;
 
         public Image playerPositionIcon;
+
+        // Used as a reference of the coordinates origin in-map and as a parcel width/height reference
+        public RectTransform centeredReferenceParcel;
+
         public MapSceneIcon scenesOfInterestIconPrefab;
 
         private HashSet<MinimapMetadata.MinimapSceneInfo> scenesOfInterest = new HashSet<MinimapMetadata.MinimapSceneInfo>();
         private Dictionary<MinimapMetadata.MinimapSceneInfo, GameObject> scenesOfInterestMarkers = new Dictionary<MinimapMetadata.MinimapSceneInfo, GameObject>();
 
+        private bool parcelHighlightEnabledValue = false;
+        public bool parcelHighlightEnabled
+        {
+            set
+            {
+                parcelHighlightEnabledValue = value;
+                parcelHighlightImage.gameObject.SetActive(parcelHighlightEnabledValue);
+            }
+            get
+            {
+                return parcelHighlightEnabledValue;
+            }
+        }
+
+        public static System.Action<int, int> onParcelClicked;
```

Comment:
```suggestion
        public static System.Action<int, int> OnParcelClicked;
```

Reply:
This needs to be static? `MapRenderer` is already a singleton.

Reply:
I made that event static because when the NavmapView subcribes to it, the `MapRenderer.i` doesn't exist yet (at `Start()` time)